### PR TITLE
SPSH-1512: add USER_ID attribute to Keycloak users

### PR DESCRIPTION
**Pull request by niklaskoopmann**
Required by NextCloud, see https://ticketsystem.dbildungscloud.de/browse/SPSH-1512

### CODE REVIEW

**Review by marode-cap — APPROVED** [review]
If this is only used for Nextcloud, maybe name the attribute differently. Otherwise fine by me

**Comment by niklaskoopmann** [comment]
> If this is only used for Nextcloud, maybe name the attribute differently. Otherwise fine by me

@marode-cap Thanks for pointing that out. I had originally planned to use it as a general attribute to be used by client-specific mappers as needed. After discussing this with @kristoff-kiefer last week we decided to keep this and the ItsLearning ID attribute client-specific since this solution is only temporary and will be replaced by the Keycloak plugin we are desperately waiting to find time to implement...

Long story short, I renamed it now.
